src/easyscience/Objects/variable/descriptor_array.py

```diff
@@ -0,0 +1,760 @@
+        try:
+            new_unit = sc.Unit(unit_str)
+        except UnitError as message:
+            raise UnitError(message) from None
+
+        # Save the current state for undo/redo
+        old_array = self._array
+
+        # Perform the unit conversion
+        try:
+            new_array = self._array.to(unit=new_unit)
+        except Exception as e:
+            raise UnitError(f"Failed to convert unit: {e}") from e
+
+        # Define the setter function for the undo stack
+        def set_array(obj, scalar):
+            obj._array = scalar
+
+        # Push to undo stack
+        self._global_object.stack.push(
+            PropertyStack(self, set_array, old_array, new_array, text=f"Convert unit to {unit_str}")
+        )
+
+        # Update the array
+        self._array = new_array
+
+    def __copy__(self) -> DescriptorArray:
+        """
+        Return a copy of the current DescriptorArray.
+        """
+        return super().__copy__()
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the DescriptorArray, showing its name, value, variance, and unit.
+        Large arrays are summarized for brevity.
+        """
+        # Base string with name
+        string = f"<{self.__class__.__name__} '{self._name}': "
+
+        # Summarize array values
+        values_summary = np.array2string(
+            self._array.values, 
+            precision=4, 
+            threshold=10,  # Show full array if <=10 elements, else summarize
+            edgeitems=3,   # Show first and last 3 elements for large arrays
+        )
+        string += f"values={values_summary}"
+
+        # Add errors if they exists
+        if self._array.variances is not None:
+            errors_summary = np.array2string(
+                self.error, 
+                precision=4, 
+                threshold=10, 
+                edgeitems=3,
+            )
+            string += f", errors={errors_summary}"
+
+        # Add unit
+        obj_unit = str(self._array.unit)
+        if obj_unit and obj_unit != "dimensionless":
+            string += f", unit={obj_unit}"
+
+        string += ">"
+        string=string.replace('\n', ',')
+        return string    
+
+
+    def as_dict(self, skip: Optional[List[str]] = None) -> Dict[str, Any]:
+        """
+        Dict representation of the current DescriptorArray.
+        The dict contains the value, unit and variances, in addition
+        to the properties of DescriptorBase.
+        """
+        raw_dict = super().as_dict(skip=skip)
+        raw_dict['value'] = self._array.values
+        raw_dict['unit'] = str(self._array.unit)
+        raw_dict['variance'] = self._array.variances
+        return raw_dict
+    
+    def _smooth_operator(self,
+                         other: Union[DescriptorArray, DescriptorNumber, list, numbers.Number],
+                         operator: str,
```

Comment:
This, again, is confusing - you are using Python's `operator` function, but renamed it to `op`, and here you have a string (or a function pointer?)

Reply:
I forgot to update the type hint, it's supposed to be a `Callable`. Fixed